tracing/setup.go

```diff
@@ -25,24 +26,38 @@ import (
 	"knative.dev/pkg/tracing/config"
 )
 
-// SetupStaticPublishing sets up trace publishing for the process. Note that other
-// pieces still need to generate the traces, this just ensures that if generated, they are collected
+type Tracer interface {
+	// Shutdown allows for proper cleanup/flush.
+	Shutdown(ctx context.Context) error
+}
+
+// SetupPublishingWithStaticConfig sets up trace publishing for the process. The caller should call .Shutdown() on
+// the returned Tracer before shutdown to make sure all traces are properly flushed.
+// Note that other pieces still need to generate the traces, this just ensures that if generated, they are collected
 // appropriately. This is normally done by using tracing.HTTPSpanMiddleware as a middleware HTTP
 // handler. The configuration will not be dynamically updated.
-func SetupStaticPublishing(logger *zap.SugaredLogger, serviceName string, cfg *config.Config) error {
+func SetupPublishingWithStaticConfig(logger *zap.SugaredLogger, serviceName string, cfg *config.Config) (Tracer, error) {
 	oct := NewOpenCensusTracer(WithExporter(serviceName, logger))
 	if err := oct.ApplyConfig(cfg); err != nil {
-		return fmt.Errorf("unable to set OpenCensusTracing config: %w", err)
+		return nil, fmt.Errorf("unable to set OpenCensusTracing config: %w", err)
 	}
-	return nil
+	return oct, nil
```

Comment:
I don't think so. Components that are creating the tracer will do some work and they need to shutdown the tracer at the end (before their own shutdown). And they do it by calling Shutdown throught the tracing.Tracer interface. So the components will now depend only on the pkg/tracing package.
If you have other thoughts how this could be done please let me know. 
There's not https://github.com/knative/eventing/pull/6474 which shows how this would be used.